File: web3rpc/rpc-specs/code-samples/python/kaia/block/getCFS.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

w3 = Web3(Web3.HTTPProvider(host))

# kaia_getCFS has no dedicated SDK wrapper; call it through the provider.
kaia_response = w3.provider.make_request("kaia_getCFS", ["latest"])

print(kaia_response)

File: web3rpc/rpc-specs/code-samples/javascript/kaia/block/getCFS.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(async () => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");

  // kaia_getCFS has no dedicated SDK wrapper; call it through the provider.
  const blockNumberOrTag = "latest";
  const result = await provider.send("kaia_getCFS", [blockNumberOrTag]);
  console.log(result);
})();
